client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -12,33 +14,35 @@ interface Params< FlowSteps extends StepperStep[] > {
 }
 
 export const useStepNavigationWithTracking = ( {
-	flow,
+	flow: { useTracksEventProps, useStepNavigation, name: flowName, variantSlug },
```

Comment:
I needed to destructure the flow here to get TS engine to flag the hook violation (from calling `flow.useTracksEventProps` inside the submit closure). 

I am versed with this rule, yet it slipped through to live (and potentially to production). I do rely on the editor/TS server to flag it. Not sure if anything we need to address, from DevEx side, if it surfaces elsewhere, etc.

Reply:
Isn't that rule an ESLint rule? I'm not sure Typscript understands React and React hooks.

Reply:
Yes, ESLint - got my terminology mixed up! Sorry.

Surprisingly, it isn't picking this up when called through the enclosing `flow` object.

Contrary to this, it does flag it when the object is imported. Like in plans data-store, we do:

```
import { Plans } from "@automattic/data-stores";

...
// then in some closure:
submit() {
    const foo = Plans.useFoo(); // flags the violation
...
```